src/Runtime/Distributions/Automata/Automaton.cs

```diff
@@ -2581,34 +2545,34 @@ private void SetStates(IEnumerable<State> newStates, int newStartStateIndex)
         /// </summary>
         /// <param name="statesToAdd">The states to add.</param>
         /// <param name="group">The group for the transitions of the states being added.</param>
-        private void AddStates(IEnumerable<State> statesToAdd, int group = 0)
+        private void AddStates(IEnumerable<StateData> statesToAdd, int group = 0)
         {
             Debug.Assert(statesToAdd != null, "A valid state collection must be provided.");
 
-            int startIndex = this.states.Count;
-            var statesToAddList = statesToAdd as IList<State> ?? statesToAdd.ToList();
+            int startIndex = this.statesData.Count;
+            var statesToAddList = statesToAdd as IList<StateData> ?? statesToAdd.ToList();
 
-            // Add states
+            // Add statesData
             for (int i = 0; i < statesToAddList.Count; ++i)
             {
                 State newState = this.AddState();
                 newState.EndWeight = statesToAddList[i].EndWeight;
 
-                Debug.Assert(statesToAddList[i].Index == i && newState.Index == i + startIndex, "State indices must always be consequent.");
+                // Debug.Assert(statesToAddList[i].Index == i && newState.Index == i + startIndex, "State indices must always be consequent.");
```

Comment:
why is this particular assertion gone?

Reply:
Because Index property is gone from StatesData. But it removes only half of the assertion. I returned the second half